### ISSUE TITLE
ArrayAs setting not working

### ISSUE THREAD

**Issue by codertimu** [zijianhuang/openapiclientgen#41]
Take the following example api spec:
```yaml
openapi: 3.0.3
info:
  title: Test openapi spec file for dictionary field
  version: v1
paths:
  '/hello':
    get:
      responses:
        '200':
          description: Success
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/TestModel'

components:
  schemas:
    TestModel:
      description: Model information
      properties:
        myArray:
          type: array
          items:
            type: string
          nullable: true

```

```json
{
  "ClientNamespace": "MyNS",
  "ClientLibraryProjectFolderName": "./TestModels",
  "CreateFolder": true,
  "GenerateModelsOnly": true,
  "ArrayAs": 6
}
```

Using `ArrayAs` setting does not change the type of the array, it is always the regular array type:
```c#
namespace MyNS
{
	
	
	/// <summary>
	/// Model information
	/// </summary>
	public class TestModel
	{
		
		[System.Text.Json.Serialization.JsonPropertyName("myArray")]
		public string[] MyArray { get; set; }
	}
}
```

**Comment by Kim-SSi** [comment]
In Fonlow.OpenApiClientGen.ClientTypes:
It looks like TypeRefHelper.CreateArrayTypeReference does not handle settings.ArrayAs.
An example is ComponentsHelper.CreateArrayOfCustomTypeReference

**Comment by zijianhuang** [comment]
fixed. to be released in 2.3.2
